isic_challenge_scoring/classification.py

```diff
@@ -61,9 +62,19 @@ def __init__(
         )
 
         self.overall = self.aggregate.at['balanced_accuracy']
-        self.validation = metrics.balanced_multiclass_accuracy(
-            truth_probabilities, prediction_probabilities, truth_weights.validation_weight
-        )
+
+        if validation_metric == 'balanced_accuracy':
+            self.validation = metrics.balanced_multiclass_accuracy(
+                truth_probabilities, prediction_probabilities, truth_weights.validation_weight
+            )
+        elif validation_metric == 'ap':
+            self.validation = metrics.average_precision(
```

Comment:
[`metrics.average_precision` takes arguments](https://github.com/ImageMarkup/isic-challenge-scoring/blob/master/isic_challenge_scoring/metrics.py#L206) of `pd.Series`, where each is the probabilities for a single category.

Here in the code, `truth_probabilities` is a `pd.DataFrame`, holding probabilities for all categories. You can reference `truth_probabilities[category]` to get a single category (see line 34 above).

I'm not sure why the type checking tests aren't catching this...

Reply:
For `auc` and `ap` should I just average the per category scores?

Reply:
I think so. For the cases where `auc` and `ap` are used (past challenges), they are mostly single-category tasks, so the average will be a no-op. In one place (2017 Task 3), there are two categories, though it's up to our discretion what specifically to choose for the validation metric.

Be sure to [compute the macro average, not the micro average](https://datascience.stackexchange.com/a/16001), since that's what we do for most of the other averages.